packages/config/src/options.ts

```diff
@@ -419,6 +419,12 @@ const resolvedOptions: Array<ResolvedConfigOption> = [
     canUpdateDuringTestTime: false,
     requireRestartOnChange: 'server',
   },
+  {
+    name: 'specPattern',
+    defaultValue: (options: Record<string, any> = {}) => options.testingType === 'component' ? defaultSpecPattern.e2e : undefined,
+    validation: validate.isStringOrArrayOfStrings,
+    canUpdateDuringTestTime: false,
+  },
```

Comment:
Am I missing something, this seems the opposite of what we want:

```js
options.testingType === 'component' ? defaultSpecPattern.e2e
```

If we are using component testing, default to the e2e spec pattern?

I think you want these: https://github.com/cypress-io/cypress/blob/50cc582543f714fd363eb785194bd1de638dd48d/packages/config/src/options.ts#L109-L112

Maybe it should be

```js
defaultValue: (options: Record<string, any> = {}) =>defaultSpecPattern[options.testingType],
```

Reply:
Yeah that was wrong... I ended up removing the defaultValue because it was causing issues. It seems like `options.testingType` was undefined sometimes when inside of the `defaultValue` function. I don't think that we need that property here based on the tests